File: SC/backend/Data/Entities/Internship.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Shared.Enums;

namespace backend.Data.Entities;


public class Internship : EntityBase
{
   [MaxLength(255)]
   [Required]
   public required string Title { get; set; }
   
   [Required]
   public required DurationType Duration { get; set; }
   
   [Required]
   [MaxLength(1000)]
   public required string Description { get; set; }
   
   [Required]
   public required DateOnly ApplicationDeadline { get; set; }
   
   [MaxLength(255)]
   [Required]
   public required string Location { get; set; }
   
   public JobCategory? JobCategory { get; set; }
   
   public JobType? JobType { get; set; }

   [MaxLength(255)] 
   public List<string> Requirements { get; set; } = new List<string>();

   [ForeignKey("CompanyId")]
   public required int CompanyId { get; set; }
   public required Company Company { get; set; }

   public List<Application> Applications { get; set; } = new List<Application>();
   
   public List<InternshipQuestion> InternshipQuestions { get; set; } = new List<InternshipQuestion>(); 
   
   public List<Match> Matches { get; set; } = new List<Match>();
}

File: SC/frontend/src/models/internship/internship.ts
import { QuestionType } from '../company/company.ts';
import { Feedback } from '../feedback/feedback.ts';

export interface Internship {
  id: number;
  dateCreated: Date;
  title: string;
  duration: number;
  description: string;
  applicationDeadline: Date;
  location: string;
  jobCategory: JobCategory;
  jobType: JobType;
  requirements: string[];
  numberOfApplications: number;
  companyId: string;
  companyName: string;
  feedbacks: Feedback[];
}

export enum JobCategory {
  Technology,
  Finance,
  Healthcare,
  Education,
  Marketing,
  Sales,
  HumanResources,
  Engineering,
  Legal,
  Operations,
}

export enum JobType {
  FullTime,
  PartTime,
}

export enum SkillsType {
  FrontEnd,
  BackEnd,
  FullStack,
  Mobile,
  DevOps,
  DataScience,
  MachineLearning,
  ArtificialIntelligence,
  CyberSecurity,
  CloudComputing,
  BlockChain,
}
export interface ApplyToInternshipInput {
  studentId?: string;
  internshipId?: string;
}

export interface AddInternshipDto {
  JobDetails: AddJobDetailsDto;
  Questions: AddQuestionDto[];
  ExistingQuestions: number[];
}

export interface AddJobDetailsDto {
  Title: string;
  Duration: DurationType;
  Description: string;
  ApplicationDeadline: string;
  Location: string;
  JobCategory?: JobCategory;
  JobType?: JobType;
  Requirements: string[];
}

export enum DurationType {
  TwoToThreeMonths = 1,
  ThreeToSixMonths = 2,
  SixToTwelveMonths = 3,
  MoreThanOneYear = 4,
}

export interface AddQuestionDto {
  Title: string;
  QuestionType: QuestionType;
  Options: string[];
}

export interface ApplicantInfo {
  skills: string[];
  name: string;
  feedbacks?: Feedback[];
  answers: ApplicantResponse[];
  studentId: string;
}

export interface ApplicantResponse {
  question: QuestionResponse;
  answer: string[];
}

export interface QuestionResponse {
  questionId: number;
  title: string;
  questionType: QuestionType;
  options: string[];
}
